operators/src/pro/cache/tile_cache.rs

```diff
@@ -29,16 +30,425 @@ pub struct TileCacheBackend {
+            );
+            return Err(super::error::CacheError::TileExpiredBeforeInsertion);
+        }
+
+        let tile_size_bytes = tile.byte_size();
+
+        // check if landing zone has enough space, otherwise abort caching the query
+        if !self.landing_zone_size.can_fit_bytes(tile_size_bytes) {
+            debug!(
+                "Not enough space in landing zone. Removing query {}. Landing zone size: {}. Landing zone used: {}. Landing zone used percentage: {}. Tile size: {}",
+                query_id, self.landing_zone_size.total_byte_size(), self.landing_zone_size.byte_size_used(), self.landing_zone_size.size_used_fraction(), tile_size_bytes
+            );
+            self.remove_query_from_landing_zone(key, query_id);
+            return Err(super::error::CacheError::NotEnoughSpaceInLandingZone);
+        }
+
+        let (landing_zone, landing_zone_size) = self
+            .landing_zone_and_size_mut(key)
+            .ok_or(super::error::CacheError::QueryNotFoundInLandingZone)?;
+
+        let entry = landing_zone
+            .get_mut(query_id)
+            .ok_or(super::error::CacheError::QueryNotFoundInLandingZone)?;
+
+        entry.query.spatial_bounds.extend(&tile.spatial_partition()); // since the source should only produce tiles that intersect with the query, we can extend the query bounds
+        entry.query.time_interval = entry
+            .query
+            .time_interval
+            .union(&tile.time)
+            .expect("time of tile must overlap with query");
+
+        T::insert_tile(&mut entry.tiles, tile)?;
+        landing_zone_size.try_add_bytes(tile_size_bytes)?;
+
+        trace!(
+            "Inserted tile for query {} into landing zone. Landing zone size: {}. Landing zone size used: {}. Landing zone used percentage: {}",
+            query_id, self.landing_zone_size.total_byte_size(), self.landing_zone_size.byte_size_used(), self.landing_zone_size.size_used_fraction()
+        );
+
+        Ok(())
+    }
+
+    /// Finish the query and make the tiles available in the cache
+    ///
+    /// # Panics
+    ///
+    /// Panics if there is aleady a cache entry for the query. This should never happen.
+    ///
+    pub fn finish_inserting_query(
+        &mut self,
+        key: &CanonicOperatorName,
+        query_id: &QueryId,
+    ) -> Result<(), CacheError> {
+        // TODO: maybe check if this cache result is already in the cache or could displace another one
+
+        // this should always work, because the query was inserted at some point and then the cache entry was created
+        let (landing_zone, landing_zone_size) = self
+            .landing_zone_and_size_mut(key)
+            .ok_or(super::error::CacheError::QueryNotFoundInLandingZone)?;
+
+        let active_query = landing_zone
+            .remove(query_id)
+            .ok_or(super::error::CacheError::QueryNotFoundInLandingZone)?;
+
+        // remove the size of the query from the landing zone
```

Comment:
maybe put it next to the evict_all_elements... method so that the size computations happens all in one place?